PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib-java/src/main/java/com/microsoft/azure/toolkit/intellij/connector/completion/provider/MethodCompletionProvider.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.intellij.connector.completion.provider;
+
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.psi.*;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.ProcessingContext;
+import com.microsoft.azure.toolkit.intellij.connector.Connection;
+import com.microsoft.azure.toolkit.intellij.connector.completion.CompletionMetadataManager;
+import com.microsoft.azure.toolkit.intellij.connector.completion.model.MethodIdentifier;
+import com.microsoft.azure.toolkit.intellij.connector.completion.model.ParameterIdentifier;
+import com.microsoft.azure.toolkit.lib.common.model.AzResource;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MethodCompletionProvider extends AbstractResourceConnectionCompletionProvider {
+
+    private static final Map<ParameterIdentifier, MethodCompletionMetadata> COMPLETION_METADATA = new HashMap<>();
+
+    static {
+        // fill meta data
+        COMPLETION_METADATA.putAll(CompletionMetadataManager.getInputProviders().stream()
+                .flatMap(provider -> provider.getMethodCompletionMetadata().stream())
+                .collect(Collectors.toMap(MethodCompletionMetadata::getIdentifier, metadata -> metadata)));
+    }
```

Comment:
prevent using static blocks, use `Service` instead to load metadata lazily to prevent slowing down startup.